src/backend/gl/src/native.rs

```diff
@@ -13,7 +13,7 @@ pub type Shader = gl::types::GLuint;
 pub type Program = gl::types::GLuint;
 pub type FrameBuffer = gl::types::GLuint;
 pub type Surface = gl::types::GLuint;
-pub type Texture = gl::types::GLuint;
+pub type Texture = (gl::types::GLenum, gl::types::GLuint);
```

Comment:
There is something wrong about this. The first piece of that tuple - a type of the texture - should be associated with the texture itself. I.e. be a member of the `ImageLike::Texture` variant. It would be cleaner than pretending that the type is a part of the identifier.

Reply:
So just to be clear, you're suggesting that the gl type be part of the gl backend's `ImageView::Texture` variant?  If so, that makes perfect sense to me, and shouldn't be very difficult.

Reply:
It was slightly more involved than I thought, but it still wasn't bad.